lib/tasks/diff_salesforce.rake

```diff
@@ -0,0 +1,61 @@
+namespace :salesforce do
+
+  namespace :schema do
+    desc "Save the salesforce schema for the given app"
+    task :dump, [:app_name] => [:environment] do |_t, args|
+      env = ENV
+      outfile = 'salesforce-schema.json'
+
+      if args[:app_name].present?
+        env = Dotenv::Parser.call(`heroku config --app #{args[:app_name]} --shell`)
+        raise "Error fetching heroku config for #{args[:app_name]}" unless $?.success?
+        outfile = "salesforce-schema-#{args[:app_name]}.json"
+      end
+
+      description = Salesforce::ApiAdapter.describe(env).group_by {|e| e["name"] }
+      description.each do |k, v|
+        raise "Expected name to be unique" unless v.one?
+        description[k] = v.first
+        description[k].delete("name")
+      end
+      File.write(
+        outfile,
+        JSON.pretty_generate(description)
+      )
+    end
+
+    desc "Diff two schema files against one another"
+    task :diff, [:old, :new] => [:environment] do |_t, args|
+      old_h = JSON.parse(File.read(args[:old]))
+      new_h = JSON.parse(File.read(args[:new]))
+
+      removed = old_h.keys - new_h.keys
+      added = new_h.keys - old_h.keys
+      diff = false
+      removed.each do |r|
+        puts "Removed object: #{r}"
+        diff = true
+      end
+      added.each do |a|
+        puts "Added object: #{a}"
+        diff = true
+      end
+
+      HashDiff.diff(old_h.except(*removed), new_h.except(*added)).each do |sym, key, old, new_val|
+        diff = true
+        case sym
+        when "~"
+          puts "Changed #{key} from #{old} to #{new_val}"
+        when "-"
+          puts "Removed #{key}"
```

Comment:
I'd have thought `-` and `+` are standard enough diff prefixes to just use them as-is?
Also, not sure if the output would be nicer if it were sorted first? (i.e. all removals, then all additions, etc). I guess changes are generally small so probably doesn't matter.

Reply:
RE `+/-/~`: It's nice to be able to copy-paste to a salesforce specialist on the other side, who might not be familiar with `diff`.

I've sorted table addition/removal but not fields (yet) - but yes, changes are generally pretty small.